lib/client/https_client.go

```diff
@@ -72,6 +72,45 @@ type WebClient struct {
 	*roundtrip.Client
 }
 
+// PostJSONWithFallback serializes an object to JSON and attempts to execute a POST
+// using HTTPS, and then fall back to plain HTTP under certain, very specific circumstances.
+//  * The caller must specifically allow it via the allowHTTPFallback parameter, and
+//  * The target host must resolve to the loopback address.
+// If these conditions are not met, then the plain-HTTP fallback is not allowed,
+// and a the HTTPS failure will be considered final.
+//
+func (w *WebClient) PostJSONWithFallback(ctx context.Context, endpoint string, allowHTTPFallback bool, val interface{}) (*roundtrip.Response, error) {
```

Comment:
Order signature so the param order matches PostJSON until it can, with additional params in the end?

```suggestion
func (w *WebClient) PostJSONWithFallback(ctx context.Context, endpoint string, val interface{}, allowHTTPFallback bool) (*roundtrip.Response, error) {
```

Reply:
Done